### PR TITLE
✨ Check if there any managed clusters exists before hub cleanup

### PR DESCRIPTION
**Pull request by RokibulHasan7**
[comment]: # ( Copyright Contributors to the Open Cluster Management project )

## Summary
If any managed clusters exist, `clusteradm clean` hangs for some time and then returns 'cluster manager still exists'. Therefore, we should check if any managed clusters exist before waiting for the hub cleanup and return the proper message.

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/cmd/clean/exec.go [hunk 1]
```diff
@@ -63,6 +65,17 @@ func (o *Options) Run() error {
...
 	}
 
+	// check if any managed cluster exist or not
+	exist, err := isManagedClusterExist(config)
+	if err != nil {
+		return err
+	}
+
+	if exist {
+		fmt.Fprintf(o.Streams.Out, "Please detach the managed clusters from the hub control plane\n")
```
Comment (zhujian7) [inline pkg/cmd/clean/exec.go:75]: Is it better to print the specific managed clusters' name? or `Please detach **all** managed clusters ...`

Comment (RokibulHasan7) [reply pkg/cmd/clean/exec.go:75]: I think `all managed clusters` is better.

File: pkg/cmd/clean/exec.go [hunk 2]
```diff
@@ -200,3 +213,23 @@ func puregeOperator(client kubernetes.Interface, extensionClient apiextensionscl
...
+		return false, err
+	}
+	managedClusters, err := clusterClient.ClusterV1().ManagedClusters().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	for _, cluster := range managedClusters.Items {
+		for _, cond := range cluster.Status.Conditions {
+			if cond.Type == "ManagedClusterConditionAvailable" && cond.Status == "True" {
```
Comment (qiujian16) [inline pkg/cmd/clean/exec.go:229]: you can use meta.IsStatusConditionTrue instead.

File: test/e2e/util/helper.go [hunk 3]
```diff
@@ -36,6 +36,29 @@ func WaitNamespaceDeleted(restcfg *rest.Config, namespace string) error {
...
+	return wait.PollUntilContextCancel(context.TODO(), 1*time.Second, true, func(ctx context.Context) (bool, error) {
+		managedCluster, err := klusterClient.ClusterV1().ManagedClusters().Get(context.TODO(), clusterName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range managedCluster.Status.Conditions {
+			if cond.Type == "ManagedClusterConditionAvailable" && cond.Status != "True" {
```
[the remark this answers is not in the document]
Comment (RokibulHasan7) [reply test/e2e/util/helper.go]: updated

**Pull request merged**